### ISSUE TITLE
Remove migration often corrupts the model snapshot in team environments   

### ISSUE THREAD

**Issue by MikaelEliasson** [dotnet/efcore#23911]
I tried to see if this was reported but could not find an issue. I hope it's not a duplicate. 

EF Core version: 5.0.2 (Think it's on all)
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Target framework: (e.g. .NET 3.1)
Operating system: Windows
IDE: (VS2019 , latest version)

## The problem

Remove migration sometimes (quite often) corrupts the ModelSnapshot by removing too much from the snapshot. This happens when two developers create migrations in parallel and then merge these back to the development branch and someone then creates a new migration and try to revert the new one. 

EF will then apply the migration snapshot from the latest of the two parallel migrations BUT it should use the merged version of these because none of the parallel migrations contains the changes from the other one in it's migration snapshot.

The issue is then that the next migrations that's added will contain more changes than it should, and most likely break when it's applied to the database.

More specifically the issue is that you can first get this when adding a migration

![image](https://user-images.githubusercontent.com/888296/104883726-7be36080-5965-11eb-93fa-450eab70761e.png)

and then when removing the same migrations right after you can get this which obviously removes things from the snapshot that was not added by this migration.

![image](https://user-images.githubusercontent.com/888296/104883773-90275d80-5965-11eb-9a54-37c651e5ea24.png)

And then when adding another new migration after that you will get this operation that tries to add the `Text` column to the database again.

![image](https://user-images.githubusercontent.com/888296/104883933-cbc22780-5965-11eb-8ccc-55922205df2c.png)

## Detailed repro

I blogged about this in more depth https://www.mikee.se/posts/corrupted_model_snapshot_in_ef_core_20210117 and there is also a small sample project for the issue https://github.com/MikaelEliasson/EFCoreSnapshotCorruptionDemo . The blog post should explain the interesting parts of the demo.

## Workarounds

Use source control like git to remove migrations instead of using `Remove migration` in a team environment.

**Comment by bricelam** [comment]
Duplicate of #8880

I hope to address situations like this (possibly with just warning) as part of #2174
